trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/TrinoS3ProxyClient.java

```diff
@@ -115,7 +116,7 @@ public void proxyRequest(SigningMetadata signingMetadata, ParsedS3Request reques
 
         request.requestContent().contentLength().ifPresent(length -> remoteRequestHeaders.putSingle("content-length", Integer.toString(length)));
 
-        signingController.inputStreamForContent(request.requestContent(), signingMetadata, Credentials::emulated)
+        signingController.inputStreamForContent(request.requestContent(), signingMetadata)
```

Comment:
Per comment below, move the body of `signingController.inputStreamForContent` here